src/utils/coordinates/coordinateExtractors.js

```diff
@@ -240,33 +322,50 @@ const executeAndReturn = (regex, text, outputProjection, extractor = numericalEx
  * - I.e. `zufall.anders.blaumeise`
  *
  * @param {String} text The text in which we want to find coordinates
- * @param {CoordinateSystem} toProjection Projection wanted for the output coordinates
- * @returns {Number[]} Coordinates in the given order in text in the wanted projection, or
- *   `undefined` if nothing was found
+ * @returns {Promise<ExtractedCoordinate | undefined>} Coordinates in the given order in text with
+ *   information about which projection they are expressed in, or `undefined` if nothing was found
  */
-const coordinateFromString = (text, toProjection) => {
+const coordinateFromString = async (text) => {
     if (typeof text !== 'string') {
         return undefined
     }
-    // creating a config array with each entry being an object with a regex attribute
-    // and the corresponding extractor when this regex matches the input
-    return [
-        { regex: REGEX_WGS_84, extractor: wgs84Extractor },
-        { regex: REGEX_WGS_84_WITH_CARDINALS, extractor: wgs84Extractor },
-        { regex: REGEX_WGS_84_WITH_PRE_FIXED_CARDINALS, extractor: wgs84Extractor },
-        { regex: REGEX_WGS_84_WITH_MIN, extractor: wgs84Extractor },
-        { regex: REGEX_WGS_84_WITH_MIN_PREFIXED, extractor: wgs84Extractor },
-        { regex: REGEX_WGS_84_WITH_SECONDS, extractor: wgs84Extractor },
-        { regex: REGEX_WGS_84_WITH_SECONDS_PREFIXED, extractor: wgs84Extractor },
-
-        { regex: REGEX_METRIC_COORDINATES, extractor: numericalExtractor },
-        { regex: REGEX_MILITARY_GRID, extractor: mgrsExtractor },
-    ]
-        .map((config) => {
-            // going through each config and extracting the result (can be undefined if not a match)
-            return executeAndReturn(config.regex, text, toProjection, config.extractor)
-        })
-        .find((result) => Array.isArray(result)) // returning the first value that is a coordinate array (will return undefined if nothing is found)
+    const wgs84result = extractWGS84Coordinates(text)
+    if (wgs84result) {
+        return {
+            coordinateSystem: WGS84,
+            coordinate: wgs84result,
+        }
+    }
+    const lv95result = extractLV95Coordinates(text)
+    if (lv95result) {
+        return {
+            coordinateSystem: LV95,
+            coordinate: lv95result,
+        }
+    }
+    const lv03result = await extractLV03Coordinates(text)
+    if (lv03result) {
+        return {
+            // coordinates have been reframed to LV95
+            coordinateSystem: LV95,
+            coordinate: lv03result,
+        }
+    }
```

Comment:
I would do the extract without reframed and after the extract do the reframe, it would make the code easier to read understand, having a `lv03result` with LV95 projection sounds wrong.

Reply:
```javascript
const lv03Result = extractLV03Coordinates(text)
if (lv03Result) {
  const lv95Result = await reframe(..)
  return {...}
}
```